### PR TITLE
Pull in changes from up-core-api 1.5.7

### PR DESCRIPTION
**Pull request by matthewd0123**
* Add Notification UMessageType
* Add isShortForm()
* Various Fixes
* 100% Code Coverage

### CODE REVIEW

**Comment by matthewd0123** [comment]
#22

**Review by neelam-kushwah — CHANGES_REQUESTED** [review]
Bump the version to 0.1.2-dev

**Review by neelam-kushwah — COMMENTED** [review]
How are you ensuring code coverage? Is there a way we can generate reports similar to what we do in Java? Let's explore this.

File: uprotocol/uri/factory/uresource_builder.py [hunk 1]
```diff
@@ -25,18 +25,28 @@
...
         return UResource(name="rpc", instance="response", id=0)
 
-    @staticmethod
-    def for_rpc_request(method, id=None):
+    @multimethod
+    def for_rpc_request(method: Union[str, None]):
+        return UResourceBuilder.for_rpc_request(method, None)
+
+    @multimethod
+    def for_rpc_request(method: Union[str, None], id: Union[int, None] = None):
```
Comment (neelam-kushwah) [inline uprotocol/uri/factory/uresource_builder.py:49]: Specifying None in type hints is often redundant as Python allows None by default for any parameter.

**Review by neelam-kushwah — CHANGES_REQUESTED** [review]

**Review by neelam-kushwah — APPROVED** [review]

**Pull request merged**